### ISSUE TITLE
Erro de comunicação SANDBOX Cielo

### ISSUE THREAD

**Issue by carcleo** [DeveloperCielo/API-3.0-PHP#138]
Testou tentando seguir a documentação do manual de integração da CIELO.

    <?php
    
    	header( "Content-Type: text/html; charset=utf-8" );
    
    	require_once "_global/_erros/erros.ini";
    
    	$dados = array(
    		"MerchantId"  => "8292b0f0-d43a-194a-bc2c-3906d12f581b",
    		"MerchantKey" => "XOISJWFWZLNKUJOEYNJXUQETLWDPFWLNLHENRTNI",
    		"MerchantOrderId" => "1",
    		"Authenticate" => true,
    		"sandbox" => false, // Opcional - Ambiente de Testes
    		"debug" => true, // Opcional - Exibe os dados enviados na requisição para a Cielo
    		"Customer" => array(
    			"Name" => "Comprador crédito completo",
    			"Email" => "compradorteste@teste.com",
    			"Birthdate" => "1991-01-02",
    			"Address" => array(
    				"Street" => "Rua Teste",
    				"Number" => "123",
    				"Complement" => "AP 123",
    				"ZipCode" => "12345987",
    				"City" => "Rio de Janeiro",
    				"State" => "RJ",
    				"Country" => "BRA"
    			),
    			"DeliveryAddress" => array(
    				"Street" => "Rua Teste",
    				"Number" => "123",
    				"Complement" => "AP 123",
    				"ZipCode" => "12345987",
    				"City" => "Rio de Janeiro",
    				"State" => "RJ",
    				"Country" => "BRA"
    			)
    		),
    		"Payment" => array(
    			"Type" => "CreditCard",
    			"Amount" => 15700,
    			"Currency" => "BRL",
    			"Country" => "BRA",
    			"ServiceTaxAmount" => 0,
    			"Installments" => 1,
    			"Interest" => "ByMerchant",
    			"Capture" => true,
    			"Authenticate" => false,
    			"ReturnUrl" => "http://www.hotplateprensas.com.br/retornoCielo.php",
    			"SoftDescriptor" => "123456789ABCD",
    			"CreditCard" => array(
    				"CardNumber" => "1234123412341231",
    				"Holder" => "Teste Holder",
    				"ExpirationDate" => "12/2030",
    				"SecurityCode" => "123",
    				"SaveCard" => "false",
    				"Brand" => "Visa"
    			)
    		),
    
    	);
    
    	$dadosJson = json_encode( $dados );
    
    	$urlProducao = "https://api.cieloecommerce.cielo.com.br/";
    	$urlSandbox  = "https://apisandbox.cieloecommerce.cielo.com.br/";
    
    	$ch = curl_init( $urlSandbox );
    	curl_setopt( $ch, CURLOPT_CUSTOMREQUEST, "POST" );
    	curl_setopt( $ch, CURLOPT_POSTFIELDS, $dadosJson );
            curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
            curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
    	curl_setopt( $ch, CURLOPT_RETURNTRANSFER, true );
    	curl_setopt( $ch, CURLOPT_HTTPHEADER, array(
    		'Content-Type: application/json',
    		'Content-Length: ' . strlen( $dadosJson ) ) );
            curl_setopt($ch, CURLOPT_CAINFO, 'ssl/Root.crt');
            curl_setopt($ch, CURLOPT_SSLVERSION, 4);
    
    	$result = curl_exec( $ch );
    
    	print "<pre>";
            print_r ( curl_error($ch) );
    	print_r ( $result );
    	print "</pre>";
    
    ?>

 
Estou na fase bem inicial ainda.

Gostaria de saber porque não há retorno algum.

No **inspector** do *browser* **tudo certo**.

Vou conferir na pagina de retorno para ver se chegou alguma coisa, tanto 

    $_POST

quanto

    $_GET

Estão vazios.

Alguém por favor ajuda?

Após um `print_r`, sai o erro abaixo:

    TCP connection reset by peer

**Comment by deepcell** [comment]
O ambiente sandbox nunca funcionou para o meu caso, desde o inicio da minha integracao eu tenho usado o ambiente de producao. Use a sua URL de producao para testar `$urlProducao`, passe os dados do cartao de teste forncecido no manual da cielo para testar sua integracao. Vai funcionar assim, quando voce finalizar sua integracao faca um teste com um cartao real com valor de R$1,00 e tente capturar a transacao.

**Comment by joaoissamu** [comment]
Basta criar uma conta aqui (https://cadastrosandbox.cieloecommerce.cielo.com.br/) e usar o id e a chave do sandbox que funciona

**Comment by marcosalberto** [comment]
Oi @joaoissamu,

Estou passando pelo mesmo problema de forma intermitente 
utilizando uma conta existente e com um estabelecimento com o Checkout em modo sandbox.

Tem alguma dica do que pode ser?

**Comment by joaoissamu** [comment]
@marcosalberto você está usando uma conta do sandbox? Para mim resolveu quando eu usei os dados de acesso de uma conta criada no site de cadastro do sandbox (https://cadastrosandbox.cieloecommerce.cielo.com.br/)

**Comment by marcosalberto** [comment]
@joaoissamu sim, uma conta sandbox. Este endereço que passou é de Ecommerce API, estou utilizando o Checkout Cielo. 

Entrei em contato com a Cielo relatando o problema e percebi também que este repositório está relacionado a API, caso chegue em alguma resolução publico aqui.
